tools/2024-09-gitops-flux.md

```diff
@@ -0,0 +1,289 @@
+
+This document is a technical design that outlines the changes required to support Flux in Radius. For a more general outline and a more comprehensive user scenario document, please see the [Radius + GitOps Feature Spec](https://github.com/radius-project/design-notes/blob/main/tools/2024-06-gitops-feature-spec.md).
+
+## Terms and definitions
+
+<!--
+Include any terms, definitions, or acronyms that are used in
+this design document to assist the reader. They may or may not
+be part of the user-facing experience once implemented, and can
+be specific to this design context.
+-->
+
+- **GitOps**: A set of practices that utilize Git as a single source of truth for declarative infrastructure and applications.
+
+- **Flux**: GitOps tool that automates the deployment of applications and infrastructure on Kubernetes clusters.
+
+- **ArgoCD**: Another Kubernetes + GitOps tool.
+
+- **SRE**: Site Reliability Engineer.
+
+## Objectives
+
+<!--
+Describe goals/non-goals and user-scenario of this feature to understand
+the end-user goals.
+* If the feature shares the same objectives of the existing design, link
+  to the existing doc and section rather than repeat the same context.
+* If the feature has a scenario, UX, or other product feature design doc,
+  link it here and summarize the important parts.
+-->
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6689
+
+### Goals
+
+<!--
+Describe goals to define why we are doing this work, how we will make
+priority decisions, and how we will determine success.
+-->
+
+**Goal: Users can publish Radius resources to Git repositories and use Flux to deploy them to Kubernetes clusters.**
+Users of Radius who are already using Flux for their Kubernetes resources should be able to use Flux to deploy Radius environments, applications, and other resources as well. This will allow users to manage their entire infrastructure and application stack using GitOps practices.
+
+**Goal: Users of Radius + Flux can patch their applications using bicepparams.**
+Users that are running Radius applications on Kubernetes clusters using Flux should be able to patch their applications using bicepparams.
+
+### Non goals
+
+<!--
+Describe non-goals to identify something that we won’t be focusing on 
+immediately. We won’t be expending any effort on these matters. If there
+will be follow-ups after this work, list them here. If there are things
+we plan to do in the future, but are out of scope of this design, list
+them here. Provide a brief explanation on why this is a non-goal.
+-->
+
+**Non-goal: Support for other GitOps tools**
+This design is focused on supporting Flux. Support for other GitOps tools (such as ArgoCD) is out of scope for this design and will be considered in a future design. The design principles outlined here may be applicable to other GitOps tools, but the implementation details will be specific to Flux.
+
```

Comment:
another non-goal for now: supporting patching outside of .bicep or .bicepparam updates